crates/lib/src/qpu/api.rs

```diff
@@ -238,11 +237,11 @@ impl ConnectionStrategy {
         &self,
         quantum_processor_id: Option<&str>,
     ) -> Result<get_controller_job_results_request::Target, QpuApiError> {
-        match self {
-            Self::EndpointId(endpoint_id) => Ok(
+        match self.connection_strategy() {
+            ConnectionStrategy::EndpointId(endpoint_id) => Ok(
                 get_controller_job_results_request::Target::EndpointId(endpoint_id.to_string()),
             ),
-            Self::Gateway | Self::DirectAccess => Ok(
+            ConnectionStrategy::Gateway | ConnectionStrategy::DirectAccess => Ok(
                 get_controller_job_results_request::Target::QuantumProcessorId(
                     quantum_processor_id
                         .ok_or(QpuApiError::MissingQpuId)?
```

Comment:
Pairing with my other comment - as long as we're here, this should return an `Option` rather than a `Result`, and `MissingQpuId` isn't needed here. One should be able to submit the job without a `target`, because the backend does accept that in direct-access mode. Only the gateway insists on a target, and we should allow the service to return the error stating as much rather than getting the client involved.